pulp_ansible/app/tasks/collectionversion_index.py

```diff
@@ -0,0 +1,188 @@
+import logging
+from semantic_version import Version
+
+from pulp_ansible.app.models import (
+    AnsibleDistribution,
+    CrossRepositoryCollectionVersionIndex as CVIndex,
+)
+
+from pulpcore.plugin.models import RepositoryVersion
+
+# from pulpcore.plugin.tasking import dispatch
+
+
+log = logging.getLogger(__name__)
+
+
+def _is_indexable(repo, repo_v):
+    # don't bother updating the index with a distribution doesn't exist
+    # if no distro exists, remove the index
+    if not AnsibleDistribution.objects.filter(repository=repo, repository_version=repo_v).exists():
+        CVIndex.objects.filter(repository=repo, repository_version=repo_v).delete()
+        return False
+
+    if repo_v:
+        if CVIndex.objects.filter(repository_version=repo_v, repository=repo).exists():
+            return False
+    return True
+
+
+def delete_previous_repository_version_indexes(repository_version):
+    """Delete indexes for previous repository versions."""
+    repository = repository_version.repository
+    qs = RepositoryVersion.objects.filter(repository__pk=repository.pk)
+    for repo_version in qs:
+        if repo_version == repository_version:
+            continue
+        for cvi in CVIndex.objects.filter(repository_version__pk=repo_version.pk):
+            cvi.delete()
+
+
+def delete_repository_version_indexes_by_repository(repository):
+    """Delete indexes for a repository."""
+    for cvi in CVIndex.objects.filter(repository__pk=repository.pk):
+        cvi.delete()
+
+
+def update_index_for_repo_version(repository_version, is_latest):
+    """Rebuild index by repository."""
+
+    repository = repository_version.repository
+    repository.cast()
+
+    if is_latest:
+        repo_v = None
+    else:
+        repo_v = repository_version
+
+    if not _is_indexable(repository, repo_v):
+        return
+
+    cvs = repository_version.content.filter(pulp_type="ansible.collection_version")
+    cvs = [x.cast() for x in cvs]
```

Comment:
I know this is ran in a separate task now so performance isn't the biggest issue, but I'm pretty sure each `cast` call results in a separate DB query.

Maybe use cvs query for an CollectionVersion query?
```suggestion
    cvs = repository_version.content.filter(pulp_type="ansible.collection_version").values_list("pk", flat=True)
    cvs = CollectionVersion.objects.filter(pk__in=cvs)
```